subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputIntegrationTest.groovy

```diff
@@ -132,15 +132,10 @@ class StaleOutputIntegrationTest extends AbstractIntegrationSpec {
             gradle.version=1.0
         """
         and:
-        succeeds("help")
-        then:
-        // It looks like the build may have been run with a different version of Gradle
-        // The build output has been removed
-        javaClassFile("Main.class").assertDoesNotExist()
-
-        when:
         succeeds("compileJava")
+
         then:
-        javaClassFile("Main.class").assertExists()
+        // It looks like the build may have been run with a different version of Gradle
+        javaClassFile("Main.class").lastModified() != old(javaClassFile("Main.class").lastModified())
```

Comment:
This just shows that the task has been re-executed, right? Is there a way to actually ensure that the file was removed?